crates/proto/src/serialize/txt/rdata_parsers/dnskey.rs

```diff
@@ -0,0 +1,138 @@
+use crate::rr::dnssec::rdata::dnskey::DNSKEY;
+use crate::rr::dnssec::Algorithm;
+use crate::serialize::txt::{ParseError, ParseErrorKind, ParseResult};
+
+fn bit_is_set(value: u16, bit: usize) -> bool {
+    let mask = 1 << bit;
+    value & mask == mask
+}
+
+#[allow(deprecated)]
+pub(crate) fn parse<'i, I: Iterator<Item = &'i str>>(mut tokens: I) -> ParseResult<DNSKEY> {
+    let flags_str: &str = tokens
+        .next()
+        .ok_or_else(|| ParseError::from(ParseErrorKind::Message("flags not present")))?;
+    let protocol_str: &str = tokens
+        .next()
+        .ok_or_else(|| ParseError::from(ParseErrorKind::Message("protocol not present")))?;
+    let algorithm_str: &str = tokens
+        .next()
+        .ok_or_else(|| ParseError::from(ParseErrorKind::Message("algorithm not present")))?;
+
+    let flags: u16 = flags_str.parse()?;
```

Comment:
Would prefer to use `T::from_str()` over `.parse()` (especially when it needs type annotations.